### PR TITLE
[function][runtime] NPE at RuntimeSpawner

### PR DESCRIPTION
**Pull request by sijie**
*Motivation*

getFunctionStat call can happen before spanwer starts the runtime

*Changes*

Add null checks and also change the order of closing sequence in RuntimeSpawner

### CODE REVIEW

File: pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/RuntimeSpawner.java [hunk 1]
```diff
@@ -82,7 +83,8 @@ public void start() throws Exception {
             processLivenessCheckTimer.scheduleAtFixedRate(new TimerTask() {
                 @Override
                 public void run() {
-                    if (!runtime.isAlive()) {
+                    Runtime runtime = RuntimeSpawner.this.runtime;
```
Comment (srkukarni) [inline pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/RuntimeSpawner.java]: When does runtime become null?

Comment (sijie) [reply pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/RuntimeSpawner.java]: if you close the runtime first and cancel the timer later. that's why I also changed the order of closing sequence.

**Comment by sijie** [comment]
run cpp tests
run integration tests

**Comment by sijie** [comment]
run cpp tests
run integration tests

**Comment by sijie** [comment]
run integration tests

**Comment by sijie** [comment]
ping @srkukarni ?

**Pull request merged**